nordicsemi/__main__.py

```diff
@@ -571,10 +573,11 @@ def generate(zipfile,
     else:
         sd_id_list = sd_req_list
 
-    signer = Signing()
-    default_key = signer.load_key(key_file)
-    if default_key:
-        display_sec_warning()
+    if key_file is not None:
+        signer = Signing()
+        default_key = signer.load_key(key_file)
+        if default_key:
+            display_sec_warning()
```

Comment:
Currently, a warning is displayed when signing with the default key or when using the debug flag (e.g. "do not do this in production"). Perhaps a warning should be displayed when generating unsigned packages as well?

Reply:
Good point. I did not check those warnings.